Intermittent failure in CI (traits.tests.test_regression)

Observed in a CI run in the PR https://github.com/enthought/traits/pull/1281

```
[00:01:59] ======================================================================
[00:01:59] FAIL: test_subclasses_weakref (traits.tests.test_regression.TestRegression)
[00:01:59] Make sure that dynamically created subclasses are not held
[00:01:59] ----------------------------------------------------------------------
[00:01:59] Traceback (most recent call last):
[00:01:59]   File "C:\Python35\lib\site-packages\traits\tests\test_regression.py", line 175, in test_subclasses_weakref
[00:01:59]     self.assertEqual(previous_subclasses, HasTraits.__subclasses__())
[00:01:59] AssertionError: Lists differ: [<class 'traits.tests.test_enum.ExampleModel'>, <class 't[14506 chars]ts'>] != [<class 'traits.util.tests.test_record_events.TestObject'[14506 chars]ex'>]
[00:01:59] 
[00:01:59] First differing element 0:
[00:01:59] <class 'traits.tests.test_enum.ExampleModel'>
[00:01:59] <class 'traits.util.tests.test_record_events.TestObject'>
[00:01:59] 
[00:01:59] Diff is 28478 characters long. Set self.maxDiff to None to see it.
[00:01:59] 
[00:01:59] ----------------------------------------------------------------------
[00:01:59] Ran 1416 tests in 37.356s
[00:01:59] 
[00:01:59] FAILED (failures=1, skipped=7)
[00:01:59] Command exited with code 1
```

See full log here - [log.txt](https://github.com/enthought/traits/files/5128737/log.txt)

See again on Windows / Python 3.5 in an Appveyor run:
```
======================================================================
FAIL: test_subclasses_weakref (traits.tests.test_regression.TestRegression)
Make sure that dynamically created subclasses are not held
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python35\lib\site-packages\traits\tests\test_regression.py", line 175, in test_subclasses_weakref
    self.assertEqual(previous_subclasses, HasTraits.__subclasses__())
AssertionError: Lists differ: [<cla[17 chars]test_weak_ref.Eggs'>, <class 'traits.tests.tes[14495 chars]re'>] != [<cla[17 chars]test_observe.ClassWithListOfInstance'>, <class[14495 chars]ld'>]
First differing element 0:
<class 'traits.tests.test_weak_ref.Eggs'>
<class 'traits.tests.test_observe.ClassWithListOfInstance'>
Diff is 28928 characters long. Set self.maxDiff to None to see it.
```

Seen again in a Python 3.5 / Windows CI run for PR #1272.

Run link (not likely to persist for long): https://ci.appveyor.com/project/EnthoughtOSS/traits/builds/35050317/job/0osvtsxe9rrk47gg

Relevant piece:

```
======================================================================
FAIL: test_subclasses_weakref (traits.tests.test_regression.TestRegression)
Make sure that dynamically created subclasses are not held
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python35\lib\site-packages\traits\tests\test_regression.py", line 175, in test_subclasses_weakref
    self.assertEqual(previous_subclasses, HasTraits.__subclasses__())
AssertionError: Lists differ: [<cla[18 chars]est_callable.MyCallable'>, <class 'traits.obse[14494 chars]st'>] != [<cla[18 chars]est_clone.ClassWithClassAttribute'>, <class 't[14494 chars]st'>]
First differing element 0:
<class 'traits.tests.test_callable.MyCallable'>
<class 'traits.tests.test_clone.ClassWithClassAttribute'>
Diff is 28440 characters long. Set self.maxDiff to None to see it.
----------------------------------------------------------------------
```

--- a/traits/tests/test_regression.py
+++ b/traits/tests/test_regression.py
@@ -172,7 +172,11 @@ class TestRegression(unittest.TestCase):
         _create_subclass()
         _create_subclass()
         gc.collect()
-        self.assertEqual(previous_subclasses, HasTraits.__subclasses__())
+
+        # In Python < 3.6, we can end up seeing the same subclasses but in
+        # a different order, so use assertCountEqual rather than assertEqual.
+        # Ref: enthought/traits#1282.
+        self.assertCountEqual(previous_subclasses, HasTraits.__subclasses__())
 
     def test_leaked_property_tuple(self):
         """ the property ctrait constructor shouldn't leak a tuple. """

